nucleus/deploy/model_endpoint.py

```diff
@@ -57,28 +90,27 @@ def predict_batch(
         # if batches are possible make this aware you can pass batches
         # TODO add batch support once those are out
 
-        # request_ids = {}  # Dict of url -> request id
+        def single_request(request):
+            # request has keys url and args
 
-        def single_request(inner_url):
             inner_inference_request = self.client.async_request(
                 endpoint_id=self.endpoint_id,
-                url=inner_url,
+                url=request.url,
+                args=request.args,
+                return_pickled=request.return_pickled,
             )
-            return inner_url, inner_inference_request
+            if request.request_id is not None:
+                request_key = request.request_id
+            elif request.url is not None:
```

Comment:
maybe this stuff really isn't necessary :P

Reply:
wdym?

Reply:
there's a lot of logic to try to figure out a reasonable request_id if the user didn't pass one in, e.g. use the url, or the args, which seems "a bit extra"

Reply:
Oh ok, as per my other comment ([18 hours ago](https://github.com/scaleapi/nucleus-python-client/pull/232#discussion_r818259959)), can we just generate a UUID?